### DISCUSSION TITLE
How to achieve 3 level (or more) master-detail relationship

### DISCUSSION THREAD

**Discussion by ziaatrix** [serenity-is/Serenity#7131] [General]
Hi,

I have a master-detail relationship which working fine. Now using the same approach I created a 3 level relationship

Table 1 -> Table 2 -> Table 3

When opening a record from Table 2, Table 3 data is retrieved. However,  when opening a record from Table 1, it only retrieves data until Table 2, **Table 3 is null**. Any idea how to achieve 3 level (or more) master-detail relationship

**Comment by reach2rv** [comment]
what is your row structure for table 1, 2 and 3? I am using this since couple of years without any issues and no modification to serenity.

**Reply by ziaatrix** [reply]
I have already figured it out. Serenity doesn't load child records unless you add MinSelectLevel etc, or explicitly ask for it.

**Reply by lotffy03** [reply]
how you created master - detail ?
